src/features/unified-share-modal/utils/computeSuggestedCollabs.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @flow
+ * @file Suggested Collaborator utility functions
+ * @author Box
+ */
+import type { SuggestedCollab, SuggestedCollabLookup } from '../flowTypes';
+import fuzzySearch from '../../../utils/fuzzySearch';
+
+/**
+ * Function to compute suggested collaborators given a list of contacts and cached suggested collaborators.
+ * Used by input components to help augment API results with cached suggestions using a fuzzy search.
+ *
+ * @param {Array<C>} contacts Contacts returned from the server with exact matches
+ * @param {SuggestedCollabLookup} suggestedCollaborators Cached suggested collab lookup to fuzzy match on
+ * @param {string} inputValue Value the user typed into the input box searching for collabs
+ * @param {number} maxSuggestions Maximum number of items to suggest/return, default 3
+ * @param {number} minCharacters Minimum number of characters before searching, default 3
+ * @returns {Array<SuggestedCollab>, Array<C>}
+ * [0] Suggested collaborators in recommended order
+ * [1] The rest of the contacts not suggested
+ */
+export default function computeSuggestedCollabs<C: { id: number | string }>(
+    contacts: Array<C>,
+    suggestedCollabLookup: SuggestedCollabLookup,
+    inputValue: string,
+    maxSuggestions: number = 3,
+    minCharacters: number = 3,
+    maxGaps: number = 2,
+): [Array<SuggestedCollab>, Array<C>] {
+    const contactIdSet = new Set(contacts.map(c => c.id.toString()));
```

Comment:
Does it make sense to use Set as the input type for this function? Can the parent component be changed to use Set or is it something thats only needed here internally?